reactor-netty-http/src/test/java/reactor/netty/BaseHttpTest.java

```diff
@@ -0,0 +1,127 @@
+import reactor.netty.http.server.HttpServer;
+import reactor.netty.resources.ConnectionProvider;
+
+import java.net.SocketAddress;
+import java.util.function.Supplier;
+
+/**
+ * Provides utility methods for {@link HttpServer} and {@link HttpClient} creation.
+ * At the end of each test guarantees that the server is disposed.
+ *
+ * @author Violeta Georgieva
+ * @since 1.0.3
+ */
+public class BaseHttpTest {
+
+	protected DisposableServer disposableServer;
+
+	@AfterEach
+	protected void tearDown() throws Exception {
+		if (disposableServer != null) {
+			disposableServer.disposeNow();
+		}
+	}
+
+	/**
+	 * Creates {@link HttpServer} bound on a random port with wire logging enabled.
+	 *
+	 * @return a new {@link HttpServer}
+	 */
+	public static HttpServer createServer() {
+		return createServer(0);
+	}
+
+	/**
+	 * Creates {@link HttpServer} bound on the specified port with wire logging enabled.
+	 *
+	 * @param port the port to bind to
+	 * @return a new {@link HttpServer}
+	 */
+	public static HttpServer createServer(int port) {
+		return HttpServer.create()
+		                 .port(port)
+		                 .wiretap(true);
+	}
+
+	/**
+	 * Creates {@link HttpClient} with a specified port to connect to and wire logging enabled.
+	 * The default pool will be used.
+	 *
+	 * @param port the port to connect to
+	 * @return a new {@link HttpClient}
+	 */
+	public static HttpClient createClient(int port) {
+		return createClient(null, port);
+	}
+
+	/**
+	 * Creates {@link HttpClient} with a specified port to connect to and wire logging enabled.
+	 * If a {@code pool} is specified, it will be used instead of the default pool.
+	 *
+	 * @param pool the pool to be used when creating the client, if null the default pool will be used
+	 * @param port the port to connect to
+	 * @return a new {@link HttpClient}
+	 */
+	public static HttpClient createClient(ConnectionProvider pool, int port) {
+		HttpClient client = pool == null ? HttpClient.create() : HttpClient.create(pool);
+		return client.port(port)
+		             .wiretap(true);
+	}
+
+	/**
+	 * Creates {@link HttpClient} with a specified remote address to connect to and wire logging enabled.
+	 * The default pool will be used.
+	 *
+	 * @param remoteAddress a supplier of the address to connect to
+	 * @return a new {@link HttpClient}
+	 */
+	public static HttpClient createClient(Supplier<SocketAddress> remoteAddress) {
+		return createClient(null, remoteAddress);
+	}
+
+	/**
+	 * Creates {@link HttpClient} with a specified remote address to connect to and wire logging enabled.
+	 * If a {@code pool} is specified, it will be used instead of the default pool.
+	 *
+	 * @param pool the pool to be used when creating the client, if null the default pool will be used
+	 * @param remoteAddress a supplier of the address to connect to
+	 * @return a new {@link HttpClient}
+	 */
+	public static HttpClient createClient(ConnectionProvider pool, Supplier<SocketAddress> remoteAddress) {
```

Comment:
mark the `pool` parameter as `@Nullable`

Reply:
fixed